### PR TITLE
CI: gen manpage for tarball

### LINKED ISSUE #2923: Include man pages in release tarball
Could you please include a man page for each binary in the release tarball?

### PR DESCRIPTION
**Pull request by junghans**
Fix #2923

### CODE REVIEW

**Comment by junghans** [comment]
2 issues:

- [ ] `sphinx-build` is not in the image, but we could install it with `apt`
- [ ] `ci/create_source_tarball.sh` is also called in `ci/build_tmp.xsh`, so we either have to build the doc there, too, or add a conditional to script.

**Comment by certik** [comment]
Just put sphinx into `ci/environment_linux.yml`, that should take care of both issues, since sphinx should be present when the create tarball is called.

**Comment by certik** [comment]
The error is:
```
Error copying directory from "man" to "lfortran-0.29.0-308-g82f419018/man".
```

**Comment by junghans** [comment]
> * `ci/create_source_tarball.sh` is also called in `ci/build_tmp.xsh`, so we either have to build the doc there, too, or add a conditional to script.

This is "ci/create_source_tarball.sh is also called in ci/build_tmp.xsh, so we either have to build the doc there, too, or add a conditional to script."

I will think about a solution.

**Comment by junghans** [comment]
@certik do you know what this means?
```
reading sources... [ 58%] asr/asr_nodes/type_nodes/ttype

Notebook error:
NoSuchKernel in ast_and_asr.ipynb:
No such kernel named fortran
reading sources... [ 59%] ast_and_asr
```

**Comment by junghans** [comment]
Now I have the same error everywhere!

**Comment by junghans** [comment]
@certik Ok, I think go get the `fortran` kernel, we need to build the source, which might be not desirable for all tarballs.

Another solution would be to create the manpages using `help2man`, which could be done easily at build time.

**Comment by junghans** [comment]
@isuruf any ideas?

**Comment by certik** [comment]
We need to decide if the documentation must be fully build for the source tarball. For SymPy we ship the documentation separately (another tarball) since it can be quite big.

Regarding manpages, I would recommend to not tie them to the full documentation, but just generate them directly one way or the other. I would still like to write and maintain the man page as Markdown document, but we can then use any tool that can convert it.

That way the build should be quite simple and reliable, we just install the "Markdown->man page" tool, and run it. No need to build anything else.

**Comment by junghans** [comment]
Yeah that seems like a good idea.

And well, you run `ronn` or `pandoc` for the md to groff conversion during the tarball creation.

Ok, give me a markdown file and will do the rest.

**Comment by certik** [comment]
Sure, here it is: https://gist.github.com/certik/76877e5722a9819f355ee3b021cc332c

If the tool you choose requires a different format, let me know, I can convert it.

**Comment by junghans** [comment]
@certik I might went a bit overboard with add it to cmake as well, but ready for review now.

File: doc/man/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# in tarball
+if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/lfortran.1)
+  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/lfortran.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
+# in git repo
+elseif(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/lfortran.md)
+  find_program(PANDOC_EXECUTABLE NAMES pnadoc)
```
Comment (certik) [inline doc/man/CMakeLists.txt:6]: Shouldn't this be:
```suggestion
  find_program(PANDOC_EXECUTABLE NAMES pandoc)
```
How did this work at the CI?

Comment (junghans) [reply doc/man/CMakeLists.txt]: The CI does not check whether the manpage actually gets build or installed, let me change that.

Comment (junghans) [reply doc/man/CMakeLists.txt]: Done!

**Review by certik — APPROVED** [review]
I tested it locally, it seems to work!

Thank for implementing it. I am going to merge this now.

**Pull request merged**